### PR TITLE
Rm/finalize loan

### LINKED ISSUE #86: Admin route to finalize loan

### PR DESCRIPTION
**Pull request by ricardomilhazes**
Added method and route to finalize and cancel loans

Obs:
- Couldn't test finalize route
- Might be worth to add some more error responses

### CODE REVIEW

File: tuichain/api/views/loans.py [hunk 1]
```diff
@@ -116,6 +116,58 @@ def create_loan_request(request):
...
+    if loan.state != LoanState.APPROVED.value:
+        return Response(
+            {"error": "The given Loan cannot be canceled"},
+            status=HTTP_403_FORBIDDEN,
+        )
+
+    controller.loans.get_by_identifier(LoanIdentifier(loan.identifier)).cancel()
+
+    return Response(
+        {"message": "Loan has been canceled"}, status=HTTP_201_CREATED
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: Is 200 more adequate?

File: tuichain/api/views/loans.py [hunk 2]
```diff
@@ -266,6 +318,51 @@ def validate_loan_request(request, id):
...
+            {"error": "The given Loan cannot be finalized"},
+            status=HTTP_403_FORBIDDEN,
+        )
+
+    controller.loans.get_by_identifier(
+        LoanIdentifier(loan.identifier)
+    ).finalize()
+
+    return Response(
+        {"message": "Loan has been finalized"}, status=HTTP_201_CREATED
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: Is 200 more adequate?

File: tuichain/api/views/loans.py [hunk 3]
```diff
@@ -116,6 +116,58 @@ def create_loan_request(request):
...
+    user = request.user
+    loan = Loan.objects.filter(id=id).first()
+
+    if user != loan.student and not user.is_superuser:
+        return Response(
+            {"error": "Cannot cancel a loan that you don't own"},
+            status=HTTP_403_FORBIDDEN,
+        )
+
+    if loan is None:
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: This check should come before the one above.

File: tuichain/api/views/loans.py [hunk 4]
```diff
@@ -116,6 +116,58 @@ def create_loan_request(request):
     )
 
 
+@api_view(["PUT"])
```
Comment (albertofaria) [inline tuichain/api/views/loans.py:119]: Is POST more adequate?

File: tuichain/api/views/loans.py [hunk 5]
```diff
@@ -266,6 +318,51 @@ def validate_loan_request(request, id):
...
```
Comment (albertofaria) [inline tuichain/api/views/loans.py:319]: Is POST more adequate?

File: tuichain/api/views/loans.py (hunk 4 above)
Comment (ricardomilhazes) [reply tuichain/api/views/loans.py:119]: It might be, since it's meant to be a "request" to blockchain

File: tuichain/api/views/loans.py (hunk 1 above)
Comment (ricardomilhazes) [reply tuichain/api/views/loans.py]: I thought the same but we are using 201 on the others like validate and reject
Should I change them all?

**Review by albertofaria — APPROVED** [review]

**Review by mrr37 — APPROVED** [review]

**Pull request merged**